openhtf/util/console_output.py

```diff
@@ -92,14 +97,16 @@ def banner_print(msg, color='', width=60, file=sys.stdout, logger=_LOG):
     ======================== Foo Bar Baz =======================
 
   """
-  logger.debug(ANSI_ESC_RE.sub('', msg))
-  if not CLI_QUIET:
-    lpad = int(math.ceil((width - _printed_len(msg) - 2) / 2.0)) * '='
-    rpad = int(math.floor((width - _printed_len(msg) - 2) / 2.0)) * '='
-    file.write('{sep}{color}{lpad} {msg} {rpad}{reset}{sep}{sep}'.format(
-        sep=_linesep_for_file(file), color=color, lpad=lpad, msg=msg, rpad=rpad,
-        reset=colorama.Style.RESET_ALL))
-    file.flush()
+  if logger:
```

Comment:
If the intention is to use `None` to disable the logging aspect, I would kinda prefer we use explicit `is not None` checks rather than implicit truthiness checks. Someone could conceivably (though weirdly) pass in a legit logger that has a falsey `__bool__` override.

On a related note, effect of passing in `None` should probably be mentioned in the arg docstrings throughout.

Reply:
Will respond in #784